openupgrade_scripts/scripts/hr_holidays/18.0.1.6/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,114 @@
+hr_holidays  / hr.leave                 / request_hour_to (selection)   : type is now 'float' ('selection')
+# DONE: pre-migration: transform to float
+
+hr_holidays  / hr.leave                 / active (boolean)              : DEL
+hr_holidays  / hr.leave.allocation      / active (boolean)              : DEL
+hr_holidays  / hr.leave                 / state (selection)             : selection_keys added: [cancel], removed: [draft]
+hr_holidays  / hr.leave.allocation      / state (selection)             : selection_keys added: [validate1] (most likely nothing to do)
+# DONE: pre-migration: set state = 'cancel' if archived. Also, 'draft' set to 'confirm' for leaves
+
+hr_holidays  / hr.leave.accrual.level   / accrual_validity (boolean)    : NEW
+hr_holidays  / hr.leave.accrual.level   / accrual_validity_count (integer): NEW hasdefault: default
+hr_holidays  / hr.leave.accrual.level   / accrual_validity_type (selection): NEW required, selection_keys: ['day', 'month'], hasdefault: default
+# NOTHING TO DO: new functionality
+
+hr_holidays  / hr.leave.accrual.level   / cap_accrued_time_yearly (boolean): NEW
+hr_holidays  / hr.leave.accrual.level   / maximum_leave_yearly (float)  : NEW
+# NOTHING TO DO: new feature
+
+hr_holidays  / hr.leave.allocation      / accrual_plan_id (many2one)    : now a function
+# NOTHING TO DO
+
+hr_holidays  / hr.leave.allocation      / actual_lastcall (date)        : NEW
+hr_holidays  / hr.leave.allocation      / last_executed_carryover_date (date): NEW
+# NOTHING TO DO
+
+hr_holidays  / hr.leave.allocation      / carried_over_days_expiration_date (date): NEW
+hr_holidays  / hr.leave.allocation      / expiring_carryover_days (float): NEW
+# NOTHING TO DO: new functionality
+
+hr_holidays  / hr.leave.allocation      / is_name_custom (boolean)      : NEW
+hr_holidays  / hr.leave.allocation      / name (char)                   : is now stored
+hr_holidays  / hr.leave.allocation      / name (char)                   : not a function anymore
+hr_holidays  / hr.leave.allocation      / private_name (char)           : DEL
+# DONE: pre-migration: renamed private_name to name
+
+hr_holidays  / hr.leave.allocation      / number_of_hours_display (float): is now stored
+# NOTHING TO DO: let odoo compute it
+
+hr_holidays  / hr.leave.allocation      / second_approver_id (many2one) : NEW relation: hr.employee
+# NOTHING TO DO: new functionality
+
+hr_holidays  / hr.leave.allocation      / yearly_accrued_amount (float) : NEW
+# NOTHING TO DO
+
+hr_holidays  / hr.leave.type            / allocation_validation_type (selection): selection_keys added: [both, hr, manager, no_validation], removed: [no, officer]
+# DONE: pre-migration: officer -> hr, no -> no_validation
+
+hr_holidays  / hr.leave.type            / include_public_holidays_in_duration (boolean): NEW hasdefault: default
+# NOTHING TO DO: new feature
```

Comment:
Need to consider whether we should do something due to https://github.com/OCA/hr-holidays/tree/18.0/hr_holidays_public/